packages/core/docs/composition/composables.md

```diff
@@ -1,128 +1,108 @@
+## What are composable?
 
-To use a composable, you need to import it from an integration you use, and call it on the component `setup` option:
+Composables are functions with an **internal state** that changes over time and **methods** that modify this state. You cannot directly modify the state but only by calling these methods. However, because the state is reactive — thanks to the Compositions API available in Vue applications — you can watch and react to these changes when necessary to update the UI or perform other operations.
 
-```js
-import { useProduct } from '{INTEGRATION}';
-import { onSSR } from '@vue-storefront/core`
-
-export default {
-  setup() {
-    const { products, search } = useProduct();
-
-    onSSR(async () => {
-      await search(searchParams);
-    });
-
-    return {
-      products
-    };
-  }
-};
-```
-
-> `onSSR` is used to perform an asynchronous request on the server-side and convey the received data to the client. You will learn more about it next section.
-
-For some composables (like `useProduct`), you will need to pass a unique ID as a parameter (it can be a product ID, category ID, etc.). Others (like `useCart`) do not require an ID passed. You can always check a composable signature in the [API Reference](../reference/api/core.html).
+This pattern encapsulates the state and business logic and exposes it through easy-to-use methods.
 
 ## Anatomy of a composable
 
-Every Vue Storefront composable usually returns three main pieces:
+Most composables consist of one or more of the following:
 
-- **Main data object**. A read-only object that represents data returned by the composable. For example, in `useProduct` it's a `products` object, in `useCategory` it's `categories` etc.
-- **Supportive data objects**. These properties depend directly or indirectly on the main data object. For example, `loading`, `error` or `isAuthenticated` from `useUser` depend on a `user`.
-- **Main function that interacts with data object**. This function usually calls the API and updates the main data object. For example in `useProduct` and `useCategory` it's a `search` method,in `useCart` it's a `load` method. The rule of thumb here is to use `search` when you need to pass some search parameters. `load` is usually called when you need to load some content based on cookies or `localStorage`
-- **platform-specific API access**. This is the section where you can reach out to the API functions that are specific to the platform you are using. By default we provide an agnostic approach of using the same API and function calls for each platform, however sometimes there is a need to use something very specific to the certain service, thus you can easily access it over the `api` object.
+- **Primary state** - read-only state of the composable. You cannot update it directly but only by calling composable **methods**.
```

Comment:
- **Primary state** - read-only state of the composable. This cannot be updated directly.